Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 2)
+++ CMakeLists.txt	(working copy)
@@ -46,11 +46,13 @@
     )
 
 find_package(SofaGui REQUIRED)
-find_package(SofaGeneral REQUIRED)
+find_package(SofaGeneral  REQUIRED)
+find_package(SofaAdvanced REQUIRED)
+find_package(SofaMisc     REQUIRED)
 set(SOFAPHYSICSAPI_DEPENDENCY_INCLUDE_DIR ${SofaGui_INCLUDE_DIRS} ${SofaGeneral_INCLUDE_DIRS}) 
     
 add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES} )
-target_link_libraries( ${PROJECT_NAME} PUBLIC SofaGuiMain SofaComponentGeneral ) 
+target_link_libraries( ${PROJECT_NAME} PUBLIC SofaGuiMain SofaComponentGeneral SofaComponentAdvanced SofaComponentMisc ) 
 target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>") 
 target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${SOFAPHYSICSAPI_DEPENDENCY_INCLUDE_DIR}>")
 
Index: SofaPhysicsAPIConfig.cmake.in
===================================================================
--- SofaPhysicsAPIConfig.cmake.in	(revision 2)
+++ SofaPhysicsAPIConfig.cmake.in	(working copy)
@@ -7,13 +7,17 @@
 
 set_and_check(SOFAPHYSICSAPI_INCLUDE_DIR  "@PACKAGE_SOFAPHYSICSAPI_INCLUDE_DIR@")
 
-check_required_components(SofaPhysicsAPI)
+find_package(SofaGeneral  REQUIRED)
+find_package(SofaGui      REQUIRED)
+find_package(SofaGeneral  REQUIRED)
+find_package(SofaAdvanced REQUIRED)
+find_package(SofaMisc     REQUIRED)
 
-find_package(SofaGeneral REQUIRED)
-
 if( NOT TARGET SofaPhysicsAPI )
 	include("${CMAKE_CURRENT_LIST_DIR}/SofaPhysicsAPITargets.cmake")
 endif() 
 
+check_required_components(SofaPhysicsAPI)
+
 set(SofaPhysicsAPI_LIBRARIES SofaPhysicsAPI )
 set(SofaPhysicsAPI_INCLUDE_DIRS ${SOFAPYTHON_INCLUDE_DIR} ${SofaGeneral_INCLUDE_DIRS} )
\ No newline at end of file
Index: SofaPhysicsSimulation.cpp
===================================================================
--- SofaPhysicsSimulation.cpp	(revision 2)
+++ SofaPhysicsSimulation.cpp	(working copy)
@@ -24,7 +24,11 @@
 
 #include "fakegui.h"
 
+#include <SofaComponentCommon/initComponentCommon.h>
+#include <SofaComponentBase/initComponentBase.h>
 #include <SofaComponentGeneral/initComponentGeneral.h>
+#include <SofaComponentAdvanced/initComponentAdvanced.h>
+#include <SofaComponentMisc/initComponentMisc.h>
 
 #include <math.h>
 #include <iostream>
@@ -170,6 +174,14 @@
     static bool first = true;
     if (first)
     {
+        sofa::component::initComponentBase();
+        sofa::component::initComponentCommon();
+        sofa::component::initComponentGeneral();
+        sofa::component::initComponentAdvanced();
+        sofa::component::initComponentMisc();
+
+        sofa::simulation::xml::initXml();
+
         if ( !useGUI ) {
           // FakeGUI to be able to receive messages
           FakeGUI::Create();
@@ -203,9 +215,6 @@
     m_Simulation = new sofa::simulation::tree::TreeSimulation();
     sofa::simulation::setSimulation(m_Simulation);
 
-    sofa::component::initComponentGeneral();
-    sofa::simulation::xml::initXml();
-
     sofa::core::ObjectFactory::AddAlias("VisualModel", "OglModel", true,
             &classVisualModel);
 
